Assets.Scripts.Core.History/HistoryWindow.cs

```diff
@@ -57,36 +62,130 @@ public void Leave(MenuUIController.MenuCloseDelegate onClose)
 			StartCoroutine(LeaveMenuAnimation(onClose));
 		}
 
+		/// <summary>
+		/// Trims the text in the TextMeshPro object based on the given line number
+		/// A positive line number will trim anything after the given line
+		/// A negative line number will trim off the first `-line` lines
+		/// </summary>
+		/// <param name="tmp">The TextMeshPro object containing the text you want to trim</param>
+		/// <param name="line">The line number for trimming</param>
+		private void TrimTextInTMP(TextMeshPro tmp, int line)
+		{
+			tmp.ForceMeshUpdate();
+			TMP_CharacterInfo[] charInfo = tmp.textInfo.characterInfo;
+			if (tmp.textInfo.lineCount <= line - 1)
+			{
+				return;
+			}
+			if (tmp.textInfo.lineCount <= -line)
+			{
+				tmp.text = "";
+				return;
+			}
+			//Debug.Log("Trimming " + tmp.text + " to " + line + " lines");
+			int target = line;
+			if (target < 0)
+			{
+				target = -target - 1;
+			}
+			int foundChar = -1;
+			int multiCharChars = 0;
+			foreach (TMP_CharacterInfo info in charInfo)
+			{
+				//Debug.Log("Character Info: index " + info.index + " meshIndex " + info.meshIndex + " visible " + info.isVisible + " xAdvance " + info.xAdvance);
```

Comment:
Are these comments still needed?  Holding on to in case a bug happens?

On that note DoctorDiablo suggested adding a log facade by default would not print anything and so wouldn't clutter up the user logs, but that we could toggle on for development so you could leave statements like this in the code uncommented.

Reply:
Yeah mainly there for convenience when debugging those functions

They're really spammy so I'm not sure I would want them on even in a general debug mode...